### PR TITLE
[incubator/vault] Bump up appVersion, default vault's image tag to appVersion, update consul's image tag.

### PR DESCRIPTION
**Pull request by varshavaradarajan**
#### What this PR does / why we need it:

Update the vault and consul image tags and the app version. The current version is over 3 months old. Thought that the image versions should be latest or as close to latest for a while so that users don't have to override the image tag values all the time. 

#### Special notes for your reviewer:
Don't know if it was intentionally left to be `0.11.2`.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by varshavaradarajan** [comment]
/assign @unguiculus 

@unguiculus - do you mind taking a look at this when you're free? Thank you.

File: incubator/vault/templates/deployment.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ spec:
     spec:
       containers:
       - name: {{ .Chart.Name }}
-        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        image: "{{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
```
Comment (sameersbn) [inline incubator/vault/templates/deployment.yaml:28]: this is very interesting. :+1:

Comment (rimusz) [reply incubator/vault/templates/deployment.yaml:28]: yes, I use that way a lot too :)

**Pull request merged**